documentation/src/main/asciidoc/user_guide/chapter-63-Streams.adoc

```diff
@@ -0,0 +1,416 @@
+==  Streams
+
+Java 8 introduced the concept of a link:https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html[Stream]
+which allows functional-style operations on collections rather than having to procedurally
+iterate over the data yourself. Stream operations can be implemented in a fashion very
+similar to MapReduce.  Streams, just like MapReduce allow you to perform processing
+upon the entirety of your cache, possibly a very large data set, but in an efficient way.
+
+Also since we can control how the entries are iterated upon we can more efficiently perform the operations in
+a cache that is distributed if you want it to perform all of the operations across the cluster
+concurrently.
+
+A stream is retrieved from the link:https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--[entrySet],
+link:https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--[keySet] or
+link:https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#values--[values] collections returned from the
+Cache by invoking the link:https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#stream--[stream] or
+link:https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#parallelStream--[parallelStream] methods.
```

Comment:
I'm not sure about referencing the `Map` and `Collection` methods here... I'd either reference our own Javadoc, or use example code code instead, like `cache.entrySet().stream()`.